src/main.c

```diff
@@ -72,10 +87,361 @@ static int app_settings_register(struct golioth_client *client)
     return err;
 }
 
+enum golioth_status write_to_storage(uint32_t block_idx,
+                                     uint8_t *block_buffer,
+                                     size_t block_size,
+                                     bool is_last,
+                                     void *callback_arg)
+{
+    const char *filename = callback_arg;
+    struct fs_file_t fp = {};
+    fs_mode_t flags = FS_O_CREATE | FS_O_WRITE;
+    char path[32];
+    int err;
+    ssize_t ret;
+
+    LOG_INF("Writing %s block idx %u", filename, (unsigned int) block_idx);
+
+    if (block_idx == 0) {
+        flags |= FS_O_TRUNC;
+    }
+
+    sprintf(path, "/storage/%s", filename);
+
+    err = fs_open(&fp, path, flags);
+    if (err) {
+        LOG_ERR("Failed to open %s: %d", filename, err);
+
+        return GOLIOTH_ERR_FAIL;
+    }
+
+    err = fs_seek(&fp, block_idx * 1024, FS_SEEK_SET);
```

Comment:
We'll need a better way to handle this once we start negotiating the block size, but for now, we can at least use the kconfig value instead of magic number.
```suggestion
    err = fs_seek(&fp, block_idx * GOLIOTH_BLOCKWISE_DOWNLOAD_BUFFER_SIZE, FS_SEEK_SET);
```